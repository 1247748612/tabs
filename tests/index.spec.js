/** @jsx React.DOM */

var expect = require('expect.js');
var Tabs = require('../index');
var TabPane = Tabs.TabPane;
var React = require('react/addons');
// var sinon = require('sinon');
var TestUtils = React.addons.TestUtils;
var Simulate = TestUtils.Simulate;

var node = document.createElement('div');
document.body.appendChild(node);

describe('tabs', function () {
  var tabs;

  beforeEach(function (done) {
    React.render(<Tabs activeKey="2">
      <TabPane tab='tab 1' key="1">first</TabPane>
      <TabPane tab='tab 2' key="2">second</TabPane>
      <TabPane tab='tab 3' key="3">third</TabPane>
    </Tabs>, node, function () {
      tabs = this;
      done();
    });
  });

  afterEach(function () {
    React.unmountComponentAtNode(node);
  });

  it('create works', function () {
    expect(TestUtils.scryRenderedDOMComponentsWithClass(tabs, 'rc-tabs').length).to.be(1);
  });

  it('nav works', function () {
    expect(TestUtils.scryRenderedDOMComponentsWithClass(tabs, 'rc-tabs-tab').length).to.be(3);
  });

  it('default active works', function () {
    expect(tabs.state.activeKey).to.be('2');
    expect(TestUtils.scryRenderedDOMComponentsWithClass(tabs,
      'rc-tabs-tab')[1].getDOMNode().className.indexOf('rc-tabs-tab-active ') !== -1).to.be(true);
  });

  it('onChange works', function (done) {
    tabs.setProps({
      onChange: function (d) {
        expect(d).to.be('1');
        done();
      }
    }, function () {
      tabs = this;
      var tab = TestUtils.scryRenderedDOMComponentsWithClass(tabs, 'rc-tabs-tab')[0];
      Simulate.click(tab);
    });
  });
});
