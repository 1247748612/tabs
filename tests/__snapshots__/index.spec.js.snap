// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rc-tabs should render Tabs with correct DOM structure 1`] = `
<div
  className="rc-tabs rc-tabs-top"
  onScroll={[Function]}
  style={Object {}}
>
  <div
    className="rc-tabs-bar"
    onKeyDown={[Function]}
    role="tablist"
    style={Object {}}
    tabIndex="0"
  >
    <div
      className="rc-tabs-nav-container"
      key="container"
    >
      <span
        className="rc-tabs-tab-prev rc-tabs-tab-btn-disabled"
        onClick={null}
        onTransitionEnd={[Function]}
        unselectable="unselectable"
      >
        <span
          className="rc-tabs-tab-prev-icon"
        />
      </span>
      <span
        className="rc-tabs-tab-next rc-tabs-tab-btn-disabled"
        onClick={null}
        unselectable="unselectable"
      >
        <span
          className="rc-tabs-tab-next-icon"
        />
      </span>
      <div
        className="rc-tabs-nav-wrap"
      >
        <div
          className="rc-tabs-nav-scroll"
        >
          <div
            className="rc-tabs-nav rc-tabs-nav-animated"
          >
            <div>
              <div
                aria-disabled="false"
                aria-selected="false"
                className=" rc-tabs-tab"
                key="1"
                onClick={[Function]}
                role="tab"
                style={
                  Object {
                    "marginRight": null,
                  }
                }
              >
                tab 1
              </div>
              <div
                aria-disabled="false"
                aria-selected="true"
                className="rc-tabs-tab-active rc-tabs-tab"
                key="2"
                onClick={[Function]}
                role="tab"
                style={
                  Object {
                    "marginRight": null,
                  }
                }
              >
                tab 2
              </div>
              <div
                aria-disabled="false"
                aria-selected="false"
                className=" rc-tabs-tab"
                key="3"
                onClick={[Function]}
                role="tab"
                style={
                  Object {
                    "marginRight": null,
                  }
                }
              >
                tab 3
              </div>
            </div>
            <div
              className="rc-tabs-ink-bar rc-tabs-ink-bar-animated"
              key="inkBar"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="rc-tabs-content rc-tabs-content-animated"
    style={
      Object {
        "MozTransform": "translateX(-100%) translateZ(0)",
        "WebkitTransform": "translateX(-100%) translateZ(0)",
        "transform": "translateX(-100%) translateZ(0)",
      }
    }
  >
    <div
      aria-hidden="true"
      className="rc-tabs-tabpane rc-tabs-tabpane-inactive"
      role="tabpanel"
    />
    <div
      aria-hidden="false"
      className="rc-tabs-tabpane rc-tabs-tabpane-active"
      role="tabpanel"
    >
      second
    </div>
    <div
      aria-hidden="true"
      className="rc-tabs-tabpane rc-tabs-tabpane-inactive"
      role="tabpanel"
    />
  </div>
</div>
`;
